core/src/test/scala/integration/kafka/server/KRaftClusterTest.scala

```diff
@@ -811,20 +811,29 @@ class KRaftClusterTest {
         assertTrue(cluster.controllers.asScala.keySet.contains(quorumInfo.leaderId),
           s"Leader ID ${quorumInfo.leaderId} was not a controller ID.")
 
-        quorumInfo.voters.forEach { voter =>
-          assertTrue(0 < voter.logEndOffset,
-            s"logEndOffset for voter with ID ${voter.replicaId} was ${voter.logEndOffset}")
-          assertNotEquals(OptionalLong.empty(), voter.lastFetchTimestamp)
-          assertNotEquals(OptionalLong.empty(), voter.lastCaughtUpTimestamp)
+        val(voters, voterResponseValid) = TestUtils.computeUntilTrue(
+          admin.describeMetadataQuorum(new DescribeMetadataQuorumOptions)
+            .quorumInfo().get().voters()){
+          voters => voters.stream
+            .allMatch(voter => (voter.logEndOffset > 0
+              && voter.lastFetchTimestamp() != OptionalLong.empty()
+              && voter.lastCaughtUpTimestamp() != OptionalLong.empty()))
         }
 
-        assertEquals(cluster.brokers.asScala.keySet, quorumInfo.observers.asScala.map(_.replicaId).toSet)
-        quorumInfo.observers.forEach { observer =>
-          assertTrue(0 < observer.logEndOffset,
-            s"logEndOffset for observer with ID ${observer.replicaId} was ${observer.logEndOffset}")
-          assertNotEquals(OptionalLong.empty(), observer.lastFetchTimestamp)
-          assertNotEquals(OptionalLong.empty(), observer.lastCaughtUpTimestamp)
+        assertTrue(voterResponseValid, s"Atleast one voter did not return the expected state within timeout." +
+          s"The responses gathered for all the voters: ${voters.toString}")
+
+        val(observers, observerResponseValid) = TestUtils.computeUntilTrue(
+          admin.describeMetadataQuorum(new DescribeMetadataQuorumOptions)
+            .quorumInfo().get().observers()){
+          observers => observers.stream.allMatch(observer => (observer.logEndOffset > 0
```

Comment:
The slight concern I had was flakiness caused by a follower not catching up to the leader ever until this check is fired. I guess it is reasonable to assume that it will happen quickly in a test environment, but wanted to avoid having a rare flake caused by that. What do you think?